### ISSUE TITLE
Position size not fully filled when I switch position direction

### ISSUE THREAD

**Issue by SeifallahSnoussi** [kieran-mackle/AutoTrader#39]
**Describe the bug**

In backtest, when a new signal is triggered, and it has a different direction than the current open position, the order size, is not entirely filled 

**Expected behavior**
Like with tradingview pine scripts, when I switch position direction, the existing position is immediately closed and a new one is opened with the exact ordered size.

Below, a screenshot of orders history, in which I added two columns about the desired position size (**My order size**) and current balance

![image](https://user-images.githubusercontent.com/63883658/178066229-c9611afc-587c-4805-b0f9-b2abc1c55ec7.png)

**Configuration**
Position size is calculated manually in each triggered signal 
`position_size  = floor(self.broker.get_NAV()/self.data.Close[i])`

**Explanation**
In the second order, the script has ordered 1310 units, however, only 81 were executed.
Of the 1310 units, 1229 were used to close the open position and the remaining ones have been executed in the new position.

I think that this behavior comes from broker.py in lines 300 to 303

![image](https://user-images.githubusercontent.com/63883658/178065377-162c69eb-1e56-49df-9846-a2ea2beffe50.png)

I tried to manipulate my order size to close the existing one and entirely fill the new position, but I got errors of insufficient margin

Is there any way to overcome this behavior? 

Thanks,

**Comment by kieran-mackle** [comment]
The easiest way to overcome this is to first close your existing position before placing the next order, which can be achieved with a ['close' order type](https://autotrader.readthedocs.io/en/latest/broker/trading.html#close-order). Simply return the closing order along with your new order. Some pseudo-code to do this:

```
# New signal
closing_order = Order(order_type='close')
new_order = Order(direction=1, size=position_size, ...)
...
return [closing_order, new_order]
```

In doing so, your existing position will be closed out before placing your next order.

You raise a good point about the margin requirements, I will have to look into adjusting the requirements when trading in the opposite direction to an existing position.

**Comment by SeifallahSnoussi** [comment]
I applied your suggestions.

The positions were closed successfully, however, the available margin was not updated, as shown below:

![image](https://user-images.githubusercontent.com/63883658/178165716-395990bf-758e-4d3d-b8ea-07e087177bdd.png)

I got many insufficient margin errors,

The solution I found to fix that is by forcing the margin update at the last line of _add_funds() in broker.py
![image](https://user-images.githubusercontent.com/63883658/178165827-1adefd9d-e150-48fb-bbd5-d1d07fac2055.png)

**Comment by kieran-mackle** [comment]
Good pickup, and good suggestion. I will add that change in the next update. Thanks!

**Comment by kieran-mackle** [comment]
This has been patched in release of version 0.7.
